Welcome Page and Transportation Method Screens

Manuella2323 opened the pull request:
🌍 **### Pull Request: Welcome Page and Transportation Method Screens**
**Summary**
This PR introduces a comprehensive welcome page and transportation method navigation system for the TripBook application. The implementation includes a welcome screen with seamless navigation to the main home screen, where users can explore different transportation options (Bus, Airplane, Train, Boat) and view corresponding travel agencies and destinations.

🔧 **Implementation Overview**
**Core Features**
🏠 Welcome Page
• Welcome Screen: Initial landing page for user onboarding
• "Let's Start" Button: Primary navigation element directing users to the home screen
• User-friendly Interface: Clean, intuitive design for first-time users
🚗 Transportation Navigation System
• Home Screen: Central hub displaying four transportation methods
• Transport Options: Bus, Airplane, Train, Boat with dedicated pages
• Agency Integration: Each transport method displays relevant travel agencies
• Destination Discovery: Places to visit organized by transportation type

🛠️ **Implementation Details**
🧩 **New Components**
WelcomeCompaniesScreen
HomeScreen
BusCompainiesScreen
AirplaneCompaniesScreen
BoatCompainiesScreen
TrainCompaniesScreen
TransportCard

**Navigation Flow**
Welcome Screen → "Let's Start" button
Home Screen → Transport method selection (Bus/Airplane/Train/Boat)
Transport-Specific Screens → Travel agencies and destinations

**ViewModel Logic**
• Navigation State Management: Handles screen transitions
• Transport Data : Organizes transport type and agencies
• Destination Management: Categorizes places by transportation method

**Visual Assets Added**
• City Images: Bafoussam, Bamenda, Bertoua, Douala, Ebolowa, Garoua, Ngaoundere, Kribi
• Agency Logos: Binam Voyage, Cam Transit, Central Voyage, General Voyage, Guaranti Express, Musango Voyage
• Amenity Icons: AC, WiFi, Charging, Entertainment, Snacks, Blanket, and default amenity icons

✅ Testing Instructions

Welcome Flow
• Launch the application
• Verify welcome screen displays correctly
• Tap "Let's Start" button
• Confirm navigation to home screen

**Transportation Navigation**
• From home screen, tap each transport method:
o Bus: Verify bus agencies and destinations display
o Airplane: Check airline companies and flight routes
o Train: Confirm railway services and train destinations
o Boat: Validate maritime transport and coastal locations

**Agency and Destination Display**
• Verify each transport screen shows:
o Relevant travel agencies with logos
o Available destinations for that transport type
o Proper categorization of services
**Navigation Consistency**
• Test back navigation from transport screens to home
• Verify seamless transitions between screens
• Confirm all buttons and interactions work properly
🚀 **Future Enhancements**
• Booking Integration: Direct booking from agency listings
• Price Comparison: Compare rates across different agencies
• Route Planning: Multi-modal transportation planning
• Implementing company and destinations details

**Screenshots**

![1](https://github.com/user-attachments/assets/0ba674e7-1406-4f52-91b5-bde82175e0a4)
![2](https://github.com/user-attachments/assets/f6dfe6b9-c07d-46d1-8f72-25cad855d758)
![3](https://github.com/user-attachments/assets/7a8b7512-56dd-4973-89ff-928ea16ab610)
![5](https://github.com/user-attachments/assets/0cce29c2-3a7d-4974-9fe0-11d1dbcd1acc)
![6](https://github.com/user-attachments/assets/811e107b-fd22-4583-abab-9abe19bd7096)

<!-- gk-ai-analysis-start:ea99db98a2e3f322884ec820cc750308fdf52334 -->
<!-- gk-ai-analysis-data:eyJzdW1tYXJ5IjoiVGhpcyBQUiBpbnRyb2R1Y2VzIGEgY29tcHJlaGVuc2l2ZSB3ZWxjb21lIHBhZ2UsIGhvbWUgc2NyZWVuLCBhbmQgbmF2aWdhdGlvbiBmbG93IGZvciBleHBsb3JpbmcgdmFyaW91cyB0cmFuc3BvcnRhdGlvbiBtZXRob2RzLCB0cmF2ZWwgYWdlbmNpZXMsIGFuZCBkZXN0aW5hdGlvbnMuIiwia2V5SW5zaWdodHMiOlt7InRpdGxlIjoiTmF2aWdhdGlvbiBhbmQgdHJhbnNwb3J0YXRpb24gZmxvdyIsImRlc2NyaXB0aW9uIjoiQWRkZWQgc2NyZWVucyBhbmQgY29tcG9uZW50cyBmb3IgZXhwbG9yaW5nIGRpZmZlcmVudCB0cmFuc3BvcnRhdGlvbiBtZXRob2RzIChCdXMsIEJvYXQsIEFpcnBsYW5lLCBUcmFpbikgYW5kIHRoZWlyIGFzc29jaWF0ZWQgYWdlbmNpZXMgYW5kIGRlc3RpbmF0aW9ucy4iLCJzZXZlcml0eSI6ImxvdyIsImZpbGVQYXRoIjoiYXBwL3NyYy9tYWluL2phdmEvY29tL2FuZHJvaWQvdHJpcGJvb2svdWkvc2NyZWVucy9CdXNDb21wYW5pZXNTY3JlZW4ua3QiLCJsaW5lU3RhcnQiOjMzfSx7InRpdGxlIjoiRmlyZWJhc2UgaW50ZWdyYXRpb24gZm9yIHRyaXBzIGRhdGEiLCJkZXNjcmlwdGlvbiI6IkludHJvZHVjZWQgYFRyaXBzUmVwb3NpdG9yeWAgdG8gZmV0Y2ggdXBjb21pbmcgdHJpcHMgYW5kIHJlY29tbWVuZGVkIHBsYWNlcyBmcm9tIEZpcmViYXNlIFJlYWx0aW1lIERhdGFiYXNlLiIsInNldmVyaXR5IjoibWVkaXVtIiwiZmlsZVBhdGgiOiJhcHAvc3JjL21haW4vamF2YS9jb20vYW5kcm9pZC90cmlwYm9vay9yZXBvc2l0b3J5L1RyaXBzUmVwb3NpdG9yeS5rdCIsImxpbmVTdGFydCI6MTR9XSwiaXNzdWVzIjpbeyJ0aXRsZSI6IlN3YWxsb3dlZCBleGNlcHRpb25zIGluIHJlcG9zaXRvcnkiLCJkZXNjcmlwdGlvbiI6IlRoZSBgZ2V0VXBjb21pbmdUcmlwc2AgYW5kIGBnZXRSZWNvbW1lbmRlZFRyaXBzYCBtZXRob2RzIGNhdGNoIGV4Y2VwdGlvbnMgYW5kIHJldHVybiBhbiBlbXB0eSBsaXN0LiBUaGlzIHByZXZlbnRzIHRoZSBlcnJvciBoYW5kbGluZyBsb2dpYyBpbiBgTWFpblZpZXdNb2RlbGAgZnJvbSBleGVjdXRpbmcsIGFzIGV4Y2VwdGlvbnMgd2lsbCBub3QgYnViYmxlIHVwIHRvIHRoZSBWaWV3TW9kZWwuIiwic2V2ZXJpdHkiOiJoaWdoIiwiZmlsZVBhdGgiOiJhcHAvc3JjL21haW4vamF2YS9jb20vYW5kcm9pZC90cmlwYm9vay9yZXBvc2l0b3J5L1RyaXBzUmVwb3NpdG9yeS5rdCIsImxpbmVTdGFydCI6NDV9XSwic3VnZ2VzdGlvbnMiOlt7InRpdGxlIjoiTWlzc2luZyBwZXJtaXNzaW9uIGhhbmRsaW5nIGNvbnRleHQiLCJkZXNjcmlwdGlvbiI6IlRoZSBgZ2V0Q3VycmVudExvY2F0aW9uYCBtZXRob2Qgc3VwcHJlc3NlcyB0aGUgbWlzc2luZyBwZXJtaXNzaW9uIHdhcm5pbmcuIFdoaWxlIHRoZSBwZXJtaXNzaW9uIGhhbmRsZXIgY29tcG9zYWJsZSBtYW5hZ2VzIGNoZWNrcyBpbiBVSSwgcmVseWluZyBvbiBzdXBwcmVzc2VkIHBlcm1pc3Npb24gYW5ub3RhdGlvbnMgaW4gdXRpbGl0eSBjbGFzc2VzIGNhbiBsZWFkIHRvIHVuZXhwZWN0ZWQgU2VjdXJpdHlFeGNlcHRpb25zIGlmIGludm9rZWQgZnJvbSBhbiB1bmF1dGhvcml6ZWQgY29udGV4dC4iLCJzZXZlcml0eSI6ImxvdyIsImZpbGVQYXRoIjoiYXBwL3NyYy9tYWluL2phdmEvY29tL2FuZHJvaWQvdHJpcGJvb2svdXRpbHMvTG9jYXRpb25VdGlscy5rdCIsImxpbmVTdGFydCI6MTV9LHsidGl0bGUiOiJMYWNrIG9mIGNhbmNlbGxhdGlvbiBzdXBwb3J0IGluIGNvcm91dGluZSBhZGFwdGVyIiwiZGVzY3JpcHRpb24iOiJUaGUgYHN1c3BlbmRDYW5jZWxsYWJsZUNvcm91dGluZWAgYmxvY2sgZm9yIGZldGNoaW5nIEZpcmViYXNlIGRhdGEgdXNlcyBhIGBWYWx1ZUV2ZW50TGlzdGVuZXJgIGJ1dCBkb2VzIG5vdCByZW1vdmUgdGhlIGxpc3RlbmVyIGlmIHRoZSBjb3JvdXRpbmUgaXMgY2FuY2VsbGVkIChlLmcuLCBpZiB0aGUgdXNlciBuYXZpZ2F0ZXMgYXdheSkuIFNpbmNlIGl0J3MgYSBTaW5nbGVWYWx1ZUV2ZW50LCB0aGUgaW1wYWN0IGlzIG1pbmltYWwsIGJ1dCBhZGRpbmcgYW4gYGludm9rZU9uQ2FuY2VsbGF0aW9uYCBoYW5kbGVyIHRvIHJlbW92ZSB0aGUgbGlzdGVuZXIgaXMgYmVzdCBwcmFjdGljZS4iLCJzZXZlcml0eSI6ImxvdyIsImZpbGVQYXRoIjoiYXBwL3NyYy9tYWluL2phdmEvY29tL2FuZHJvaWQvdHJpcGJvb2svcmVwb3NpdG9yeS9Ucmlwc1JlcG9zaXRvcnkua3QiLCJsaW5lU3RhcnQiOjI1fV0sInNlY3VyaXR5IjpbXX0= -->
<!-- gk-ai-analysis-end:ea99db98a2e3f322884ec820cc750308fdf52334 -->

<!-- gitkraken-review-badge-begin -->
---
<a href="https://gitkraken.dev/review/github/zDjangoBay/TripBook/pull/307?source=pr_review_chip">
  <picture>
    <source media="(prefers-color-scheme: dark)" srcset="https://gitkraken.dev/images/figures/gitkraken-review-badge-dark.svg">
    <img src="https://gitkraken.dev/images/figures/gitkraken-review-badge-light.svg" alt="Open with GitKraken">
  </picture>
</a>
<!-- gitkraken-review-badge-end -->

Bilibri23 committed:
Added a tripCard Ui  to preview

Bilibri23 committed:
added a proposal doc for the module

Bilibri23 committed:
deleted a file which contaoined a proposed structure

Bilibri23 committed:
stand alone view screen for trip  catalog

Bilibri23 committed:
deleted trip list directory

Bilibri23 committed:
added a model class called trip

Bilibri23 committed:
deleted the trip class , and replaced it with a more refined model class

Bilibri23 committed:
new trps

Bilibri23 committed:
replaced the dummy data (list of trips) with trip data model

Bilibri23 committed:
scaffolding my ui

levelingdev committed:
dependencies added and splash UI for our module designed and dashboard UI also members are to add code and UI and to modify the existing code which the skeleton will be provided but the splash page and dashbaord are my work

levelingdev committed:
Update README.md

levelingdev committed:
Update README.md

Manuella2323 committed:
setting up my project for future changes and to run my code properly

Manuella2323 committed:
setting up my project for future changes and to run my code properly

Bilibri23 committed:
implemented tripcatalog screen with mock data and tripcard ui

levelingdev committed:
recommended trips and upcomming schedules linked to the database for the view of the different trips.Done

Bilibri23 committed:
built filtering functionalities and adjusted the trip  card

levelingdev committed:
Revert "recommended trips and upcomming schedules linked to the database for the view of the different trips"

This reverts commit 91b30110ecad354fc40a109b9ebd93a7be4e416f.

levelingdev committed:
Reapply "recommended trips and upcomming schedules linked to the database for the view of the different trips"

This reverts commit 4c28a860341002dd39d891d66f63d1ec8284dd2a.

Manuella2323 committed:
Merge remote-tracking branch 'origin/main' into tripcatalog-gounoumireille-2

Manuella2323 committed:
Merge remote-tracking branch 'refs/remotes/origin/tripcatalog-kembeunoumbiapanthony' into tripcatalog-gounoumireille-2

Bilibri23 committed:
Removed duplicate functions

Manuella2323 committed:
Merge remote-tracking branch 'refs/remotes/origin/ngane-emmanuel-tripcatalog' into tripcatalog-gounoumireille-3

# Conflicts:
#	app/src/main/AndroidManifest.xml
#	app/src/main/java/com/android/tripbook/TripCatalogActivity.kt
#	gradle/libs.versions.toml

Manuella2323 committed:
setting up my project for future changes and to run my code properly

Manuella2323 committed:
Merge remote-tracking branch 'origin/ngane-emmanuel-tripcatalog' into tripcatalog-gounoumireille-3

# Conflicts:
#	app/src/main/java/com/android/tripbook/TripCatalogActivity.kt

Bilibri23 committed:
Accepted Ngane's version for all merge conflicts

Manuella2323 committed:
Developing the welcome page using jetpack compose

Manuella2323 committed:
Developing the welcome page using jetpack compose

Manuella2323 committed:
Merge branch 'tripcatalog-gounoumireille-3' into tripcatalog-gounoumireille

# Conflicts:
#	app/build.gradle.kts
#	app/src/main/res/values/colors.xml
#	gradle/libs.versions.toml

Bilibri23 committed:
Merge remote-tracking branch 'origin/tripcatalog-ngane-emmanuel' into tripcatalog-YohmehMarcBrian

Manuella2323 committed:
Building the welcome page which is linked to the main page by the "let's start" button

Bilibri23 committed:
integrated google maps in to the module with filtering

levelingdev committed:
save

levelingdev committed:
save

levelingdev committed:
Merge branch 'tripcatalog' of https://github.com/sas-bergson/TripBook into tripcatalog-kembeunoumbiapanthony

levelingdev committed:
save

levelingdev committed:
save changes from tripcatalog after ADB work merged

Bilibri23 committed:
restucturing the model s to adapt map integration by locations

AlainICT committed:
Merge pull request #101 from sas-bergson/tripcatalog

Tripcatalog

Rodern committed:
Update app features and user interface enhancements

- Changed `minSdk` version from 31 to 28 in `build.gradle.kts`.
- Added `votes` property to `Review` data structure in `SampleReviews.kt` for upvote/downvote functionality.
- Enhanced `ReviewCard` to display voting buttons and current vote count.
- Refactored `TripCard` to accept `miniProfileContent` for user profiles.
- Updated `TripCatalogScreen` to show user avatars in trip cards.
- Modified `MockReviewViewModel` to support voting logic, pagination, and sorting.
- Created new `User` data class in `User.kt` for user information.
- Added new composables in `MiniProfile.kt` for user profile display.
- Documented changes in a new README file.
- Added `DocumentLayout.json` for managing development layout.

levelingdev committed:
save changes from tripcatalog after ADB work merged

levelingdev committed:
save changes from tripcatalog after ADB work merged

levelingdev committed:
save changes from tripcatalog after ADB work merged

levelingdev committed:
save changes from tripcatalog after ADB work merged

levelingdev committed:
save changes from tripcatalog after ADB work merged

levelingdev committed:
save changes from tripcatalog after ADB work merged

Bilibri23 committed:
google api integraretd

Bilibri23 committed:
Merge tripcatalog into brian  - resolved TripDetailScreen.kt conflict by taking 'theirs' version

Bilibri23 committed:
added google map integration and filtering

AssonfackBraynt committed:
AddNewPlace (uncatalogued places) in to the catalogue

DjouelaMarie committed:
create a data class for comment reactions

DjouelaMarie committed:
update the Comment model to include reactions

DjouelaMarie committed:
create a component for the reaction popup

DjouelaMarie committed:
Update the CommentViewModel to handle reactions

DjouelaMarie committed:
Update the CommentCard to support reactions

DjouelaMarie committed:
Update the call to CommentsCarousel in DetailReviewScreen

DjouelaMarie committed:
Update the Comment Model to Support Replies

DjouelaMarie committed:
Create a Reply UI Component

DjouelaMarie committed:
Update the CommentViewModel to Handle Replies

DjouelaMarie committed:
Update the CommentCard to Show Replies and Reply UI

DjouelaMarie committed:
Update the CommentsCarousel to Handle Replies

levelingdev committed:
carol's code merged

Bilibri23 committed:
calculating usser current location to selected trip

Bilibri23 committed:
Merge branch 'tripcatalog-Djouelasignemarierosine-ICTU20223284-reviewreaction' of github.com:zDjangoBay/TripBook into tripcatalog-YohmehMarcBrian-ICTU20223131-mapuifilters

Bilibri23 committed:
merged reviews work to map interaction

Bilibri23 committed:
adjusted filters and fixed conflicts from merged work

levelingdev committed:
carol's code merged and working

Manuella2323 committed:
Merge remote-tracking branch 'refs/remotes/origin/tripcatalog-kembeunoumbiapanthonyjoel-ICTU20223099-HomeScreen' into tripcatalog-gounoumireille-ICTU20223326-Homemanagent

AssonfackBraynt committed:
updates the SampleTrips + adds image

levelingdev committed:
Merge branch 'tripcatalog-Djouelasignemarierosine-ICTU20223284-reviewreaction' of https://github.com/zDjangoBay/TripBook into tripcatalog-kembeunoumbiapanthonyjoel-ICTU20223099-HomeScreen

Bilibri23 committed:
resolved conflicts from add places functionality

levelingdev committed:
carol's code merged and working

Bilibri23 committed:
seemless integration of add places functionality

AssonfackBraynt committed:
retrive added places detail + merge previos works

Bilibri23 committed:
new place appears on map

Bilibri23 committed:
home screen activity integrated

levelingdev committed:
carol's code merged and working

levelingdev committed:
Update build.gradle.kts

levelingdev committed:
Update build.gradle.kts

Bilibri23 committed:
updated map to home screen

levelingdev committed:
Update build.gradle.kts

AssonfackBraynt committed:
merged anthony's update

Bilibri23 committed:
adjusted changes

Manuella2323 committed:
Modifying the welcome page which is linked to the main page by the "let's start" button. Also, creating the different pages where users are been redirected to, when clicking on the different transport methods on the home screen.

Manuella2323 committed:
Merge remote-tracking branch 'origin/tripcatalog-AssonfackBrayntRandy-ICTU20241940-AddNewPlace' into tripcatalog-gounoumireille-ICTU20223326-Homemanagent

Bilibri23 committed:
final changes

Bilibri23 committed:
fixed the trip latency issues

Manuella2323 committed:
Modifying the welcome page which is linked to the main page by the "let's start" button. Also, creating the different pages where users are been redirected to, when clicking on the different transport methods on the home screen.Continued implementing the other pages namely(airplane, boat and train pages).

Bilibri23 committed:
refactored the code to be clea, and adjusted  recomposition of cursor stare

Manuella2323 committed:
Update Trip model and add visual assets(city images, companany logo, amenity icons)

Bilibri23 committed:
double checking the cursor moving backwards

Bilibri23 committed:
all regions showing for filtering

Manuella2323 committed:
Merge remote-tracking branch 'origin/tripcatalog-YohmehMarcBrian-ICTU20223131-mapuifilters' into tripcatalog-gounoumireille-ICTU20223326-Homemanagent

Manuella2323 committed:
Merging latest required branch's code into my branch and solving naming convention issues.

Bilibri23 reviewed (approved):
Obeyed SOLID design principles, I see you followee the MVVM architecture, permitting separation of concerns good work, however , removed unused import to make code cleaner . Good work

Limalambaga reviewed (approved):
The changes are really mind blowing especially withe Cameroonian touch added to each transportation facility method. The code has conflicts (sampleTrips file missen) had to rename it before it could merge. But your code really works and it's tested just try uploading exact pictures of these transportation facilities

AssonfackBraynt reviewed (commented):
noticed a bug which closes the app when clicking in elements of BoatCompaniesScreen.kt and BusCompaniesScreen.kt

dongmo323 reviewed (approved):
The home screen is attractive suitable for the trip book

AlainICT reviewed (approved):
This pull request has significant enhancements to the TripBook project, focusing on the Welcome Page and Transportation Method Screens. The proposed changes has improved user experience, streamline navigation, and enhance the overall functionality of the platform.

The changes introduced in this pull request significantly improve the usability and functionality of the TripBook platform. The enhancements align with the project's goals and contribute to a more seamless user experience.

DjouelaMarie committed:
Fixed Comment Data Class ✅
Changed reactions from MutableMap<String, MutableList<CommentReaction>> to Map<String, List<CommentReaction>>
Changed replies from MutableList<Comment> to List<Comment>
Used emptyMap() and emptyList() as default values instead of mutable collections

Bilibri23 committed:
Merge branch 'tripcatalog-Djouelasignemarierosine-ICTU20223284-reviewreaction' of github.com:zDjangoBay/TripBook into tripcatalog-YohmehMarcBrian-ICTU20223131-mapuifilters

Bilibri23 committed:
fixed package naming convention for model, all good now

Manuella2323 committed:
Merge remote-tracking branch 'origin/tripcatalog-YohmehMarcBrian-ICTU20223131-mapuifilters' into tripcatalog-gounoumireille-ICTU20223326-Homemanagent

# Conflicts:
#	app/src/main/java/com/android/tripbook/viewmodel/MockCommentViewModel.kt

Fokoua-Will-Brayan reviewed (approved):
greate  work and well implemented

nganeemmanuelict reviewed (approved):
Good design and use of images for an immersie experience. Use of OOP was also evident and blend well with the work of other members.

GweiCarlson reviewed (commented):
great touch on the simplicity of the welcome page in terms of both the code structure/implementation as well as the user interface

levelingdev commented:
Excellent chronological addition of her work with  respect to the base design
